tests/foreman/api/test_contentviewversion.py

```diff
@@ -747,6 +750,63 @@ def test_positive_remove_cv_version_from_env_with_host_registered():
 
     :CaseLevel: System
     """
+    # Create a content view cv1
+    cv1 = entities.ContentView(organization=default_org).create()
+    # Add a yum repo to the content view
+    product = entities.Product(organization=default_org).create()
+    yum_repo = entities.Repository(
+        url=settings.repos.yum_1.url, product=product, content_type='yum'
+    ).create()
+    yum_repo.sync()
+    # add the yum repo to content view
+    cv1.repository = [yum_repo]
+    cv1 = cv1.update(['repository'])
+    # Publish the content view
+    cv1.publish()
+    # Promote the content view to multiple environment Library -> DEV -> QE
+    lce_lib = entities.LifecycleEnvironment(organization=default_org).create()
+    lce_dev = entities.LifecycleEnvironment(organization=default_org, prior=lce_lib).create()
+    lce_qe = entities.LifecycleEnvironment(organization=default_org, prior=lce_dev).create()
+    cv1 = cv1.read()
+    cv1.version[0].promote(data={'environment_ids': lce_qe.id, 'force': True})
+    # Create an Activation key with the QE environment
+    ak_lce = entities.ActivationKey(
+        organization=default_org, environment=lce_qe, content_view=cv1
+    ).create()
+    # assert created ak is associate with lce qe
+    assert ak_lce.environment.id == lce_qe.id
+
+    # Register a content-host using the Activation key
+    rhel7_contenthost.create_custom_repos(rhel7=settings.repos.rhel7_os)
```

Comment:
If I understand create_custom_repo correctly, it generates a repofile for an external repo and places it on the content host. It looks like the repo you're using here is the RHEL 7 compose repo, which is where tree is installed from a few lines down. However, that repo is not in the content view that you've created. Based on my reading of the steps and expected results, this test should install a package from the content view onto the host. Please let me know if I'm misunderstanding this.